algorithms/src/snark/marlin/ahp/indexer/indexer.rs

```diff
@@ -194,27 +191,28 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
         let non_zero_c_domain =
             EvaluationDomain::new(num_non_zero_c).ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
 
-        let (constraint_domain_elements, constraint_domain_eq_poly_vals) =
-            precomputation_for_matrix_evals(&constraint_domain);
+        let constraint_domain_elements = constraint_domain.elements().collect::<Vec<_>>();
```

Comment:
We can do an optimization that if `variable_domain == constraint_domain`, we can simply set `let variable_domain_elements = &constraint_domain_elements`. (This case happens for most real world circuits

Reply:
Nice! b1bd1e1c3